core/src/test/scala/codesearch/core/regex/space/SpaceInsensitiveSpec.scala

```diff
@@ -0,0 +1,53 @@
+package codesearch.core.regex.space
+
+import org.scalatest.{FreeSpec, Matchers}
+
+class SpaceInsensitiveSpec extends FreeSpec with Matchers {
+  def addingSpaceInsensitive(before: String, after: String) = {
+    before in {
+      SpaceInsensitive.spaceInsensitiveString(before) shouldBe after
+    }
+  }
+
+  "Space Insensitive tests" - {
+    "Simple strings" - {
+      addingSpaceInsensitive("Hello world", "Hello +world")
+
+      addingSpaceInsensitive("Hello    world", "Hello    world")
+
+      addingSpaceInsensitive("foo\\ bar", "foo\\ bar")
+    }
+
+    "Strings with regexp" - {
+      addingSpaceInsensitive("foo +bar", "foo +bar")
+
+      addingSpaceInsensitive("foo ?bar", "foo( +)?bar")
+
+      addingSpaceInsensitive("foo *bar", "foo *bar")
+    }
+
+    "Strings with other cases" - {
+      addingSpaceInsensitive("foo [bar]", "foo +[bar]")
+
+      addingSpaceInsensitive("foo {1,3}", "foo {1,3}")
+
+      addingSpaceInsensitive("foo [b ar]", "foo +[b ar]")
+    }
+
+    "*Not valid tests*" - {
+
+      /**
+        * A vot i net, this should actually become foo +{bar} because not everything in curly braces is parsed as a range.
```

Comment:
Nu vot tolko translit moj ne nado zhe bylo kopirovat 🙂 

```suggestion
        * This test is actually broken for now; this should become "foo +{bar}"
          because not everything in curly braces is parsed as a range.
```

Reply:
I did not understand that this is translit .... I also thought "what is this words...?" Ahahaha.